pkg/controller/clusterset/clusterset.go

```diff
@@ -50,14 +51,34 @@ func Add(ctx context.Context, mgr manager.Manager, clusterCIDR string, logger *l
 }
 
 func (c *ClusterSetReconciler) Reconcile(ctx context.Context, req reconcile.Request) (reconcile.Result, error) {
-	var (
-		clusterSet v1alpha1.ClusterSet
-		log        = c.logger.With("ClusterSet", req.NamespacedName)
-	)
-	if err := c.Client.Get(ctx, types.NamespacedName{Name: req.Name, Namespace: req.Namespace}, &clusterSet); err != nil {
+	log := c.logger.With("ClusterSet", req.NamespacedName)
+
+	var clusterSet v1alpha1.ClusterSet
+	if err := c.Client.Get(ctx, req.NamespacedName, &clusterSet); err != nil {
+		return reconcile.Result{}, err
+	}
+
+	// Check if a ClusterSet in the same namespace already exists.
+	// We retry in case of timeout because it could happen the Informer needs to sync
+	clusterSetList := &v1alpha1.ClusterSetList{}
+	listOpts := &ctrlruntimeclient.ListOptions{Namespace: req.Namespace}
+	err := retry.OnError(retry.DefaultBackoff, apierrors.IsTimeout, func() error {
```

Comment:
The repo standard for this return is to scope lock the error.

Reply:
Not sure what do you mean. Can you re-elaborate?

I had to add this retry because the tests were running probably too fast, and I was receiving an error about the cache not being synced yet.